src/zenml/artifacts/load_directory_materializer.py

```diff
@@ -0,0 +1,93 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Only-load materializer for directories."""
+
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, ClassVar, Tuple, Type
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+class PreexistingArtifactPath(type(Path())):  # type: ignore[misc]
```

Comment:
Does `class PreexistingArtifactPath(Path):` not work for some reason?

Reply:
Yeah, it fails to init some internal variables, like `_flavor`. I tried using `super().__init__`, but it didn't help.
Some random guys from the Internet suggest to subclass like this for `Path`.